src/ResourceManager/Sql/Commands.Sql/Elastic Jobs/Cmdlet/Job/SetAzureSqlElasticJob.cs

```diff
@@ -0,0 +1,381 @@
+            InitializeResourceIdProperties(this.ResourceId);
+            this.Name = this.Name ?? this.JobName;
+            base.ExecuteCmdlet();
+        }
+
+        /// <summary>
+        /// Check to see if the agent already exists in this resource group.
+        /// </summary>
+        /// <returns>Null if the agent doesn't exist. Otherwise throws exception</returns>
+        protected override IEnumerable<AzureSqlElasticJobModel> GetEntity()
+        {
+            try
+            {
+                return new List<AzureSqlElasticJobModel> {
+                    ModelAdapter.GetJob(this.ResourceGroupName, this.ServerName, this.AgentName, this.Name)
+                };
+            }
+            catch (CloudException ex)
+            {
+                if (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    // The job does not exist
+                    throw new PSArgumentException(
+                        string.Format(Properties.Resources.AzureElasticJobNotExists, this.Name, this.AgentName),
+                        "JobName");
+                }
+
+                // Unexpected exception encountered
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Generates the model from user input.
+        /// </summary>
+        /// <param name="model">This is null since the server doesn't exist yet</param>
+        /// <returns>The generated model from user input</returns>
+        protected override IEnumerable<AzureSqlElasticJobModel> ApplyUserInputToModel(IEnumerable<AzureSqlElasticJobModel> model)
+        {
+            var existingEntity = model.First();
+
+            AzureSqlElasticJobModel updatedEntity = new AzureSqlElasticJobModel
+            {
+                ResourceGroupName = this.ResourceGroupName,
+                ServerName = this.ServerName,
+                AgentName = this.AgentName,
+                JobName = this.Name,
+                Description = this.Description != null ? this.Description : existingEntity.Description,
+                StartTime = this.StartTime != null ? this.StartTime : existingEntity.StartTime,
+                EndTime = this.EndTime != null ? this.EndTime : existingEntity.EndTime,
+                ScheduleType = this.RunOnce.IsPresent ? JobScheduleType.Once :
+                               this.IntervalType.HasValue ? JobScheduleType.Recurring : (JobScheduleType?) existingEntity.ScheduleType,
+                Enabled = this.Enable.IsPresent
+            };
+
+            if (updatedEntity.ScheduleType != null && updatedEntity.ScheduleType == JobScheduleType.Recurring)
+            {
+                StringBuilder stringBuilder = new StringBuilder();
+
+                // Create basic ISO 8601 duration - Basic string builder implementation
+                // XmlConvert.ToString(timeSpan) only supports up to days. Weeks and months need to be supported
+                stringBuilder.Append("P");
```

Comment:
Please separate out the ISO8601 duration code. This code should be reused between Jobs and LTR.